system/Test/FilterTestTrait.php

```diff
@@ -132,26 +132,68 @@ protected function getFilterCaller($filter, string $position): Closure
                     throw new RuntimeException("No filter found with alias '{$filter}'");
                 }
 
-                $filter = $this->filtersConfig->aliases[$filter];
+                $filterClasses = $this->filtersConfig->aliases[$filter];
             }
 
-            // Get an instance
-            $filter = new $filter();
+            $filterClasses = (array) $filterClasses;
```

Comment:
This introduced a scenario where `$filterClasses` may note be set (if a FQCN was passed as `$filter` causing it to error out.

Test code:
```php
    public function testLoginPass()
    {
        session()->set('uid', '54321fedcba');

        $caller = $this->getFilterCaller(LoginFilter::class, 'before');
        $result = $caller();

        $this->assertNull($result);
    }
```

Result:
```
2) FiltersTest::testNotAuthenticated with data set #0 ('App\Filters\LoginFilter')
ErrorException: Undefined variable $filterClasses

/home/runner/work/web/web/vendor/codeigniter4/framework/system/Test/FilterTestTrait.php:138
/home/runner/work/web/web/tests/misc/FiltersTest.php:60
```

Reply:
Oh, sorry! 
Why can't static analysis detect?

Reply:
I sent #8195